core/src/main/java/com/transferwise/common/gracefulshutdown/strategies/ExecutorServiceGracefulShutdownStrategy.java

```diff
@@ -0,0 +1,57 @@
+package com.transferwise.common.gracefulshutdown.strategies;
+
+import com.transferwise.common.gracefulshutdown.config.GracefulShutdownProperties;
+import com.transferwise.common.gracefulshutdown.utils.ShutdownUtils;
+import java.time.Duration;
+import java.util.concurrent.ExecutorService;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.ApplicationContext;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+public class ExecutorServiceGracefulShutdownStrategy extends BaseReactiveResourceShutdownStrategy<ExecutorService> {
+
+  private final Duration delayBetweenTerminationCheck = Duration.ofMillis(250);
```

Comment:
Maybe we can add a new property to GracefulShutdownProperties with time interval for strategy resource termination check?